### PR TITLE
Add test cases to verify yo-BJ(Yoruba -Benin), yo-NG(Yoruba -Nigeria) locales

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [x] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
* Added test cases to verify yo-NG (Yoruba -Nigeria), yo-BJ(Yoruba -Benin)
* Added test case feature: LocaleInfo,Script, String(plural) LocaleMatch, DateFmt, DurFmt, DateRngFmt, List, Unit, firstDayofWeek, clock, calendar,Month/Weekday Translation, NumFmt,
* or translation  and some unitfmt translations are manually updated by google translate's help for ListFmt

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

**Comment by gouniLee** [comment]
@ehoogerbeets Could you please to review?

File: js/test/date/testdatefmt_yo_BJ.js [hunk 1]
```diff
@@ -0,0 +1,1694 @@
...
+            locale: "yo-BJ",
+            year: 2011,
+            month: 11,
+            day: 20,
+            hour: 13,
+            minute: 45,
+            second: 30,
+            millisecond: 0
+        });
+        test.equal(fmt.formatRelative(reference, date), "+30 s");
```
Comment (ehoogerbeets) [inline js/test/date/testdatefmt_yo_BJ.js:1327]: These relative formats do not seem right. We have translations for "seconds", "minutes", etc. in `js/data/locale/yo/sysres.json` so whey aren't we using them in the relative formats?

```
    "1#in 1 second|#in {num} seconds": "#+{num} ìṣ àáy",
    "1#in 1 minute|#in {num} minutes": "#+{num} ìṣ",
    "1#in 1 hour|#in {num} hours": "#+{num} wkt",
```

File: js/test/date/testdatefmt_yo_BJ.js [hunk 2]
```diff
@@ -0,0 +1,1694 @@
...
+            year: 2011,
+            month: 11,
+            day: 20,
+            hour: 13,
+            minute: 44,
+            second: 30,
+            millisecond: 0
+        });
+        test.equal(fmt.formatRelative(reference, date), "-30 s");
+        test.done();
```
Comment (ehoogerbeets) [inline js/test/date/testdatefmt_yo_BJ.js:1356]: Same for all the other relative formats

File: js/test/date/testdatefmt_yo_NG.js [hunk 3]
```diff
@@ -0,0 +1,1694 @@
...
+            locale: "yo-NG",
+            year: 2011,
+            month: 11,
+            day: 20,
+            hour: 13,
+            minute: 45,
+            second: 30,
+            millisecond: 0
+        });
+        test.equal(fmt.formatRelative(reference, date), "+30 s");
```
Comment (ehoogerbeets) [inline js/test/date/testdatefmt_yo_NG.js:1327]: Same here too

File: js/test/root/teststrings.js [hunk 4]
```diff
@@ -3432,6 +3432,26 @@ module.exports.teststrings = {
         test.equal(str.formatChoice(5), "Default items");
         test.done();
     },
+    testStringFormatChoiceCharClasses_yo_NG: function(test) {
+        test.expect(2);
+        var str = new IString("0#There are no items.|one#There items are one|few#The items are few|many#The items are many|#Default items");
```
Comment (ehoogerbeets) [inline js/test/root/teststrings.js:3437]: The item is one

File: js/data/locale/yo/unitfmt.json [hunk 5]
```diff
@@ -8,11 +8,11 @@
             "degree": "#{n}°",
             "arc-minute": "#{n}′",
             "arc-second": "#{n}″",
-            "square-kilometer": "#{n} km²",
-            "hectare": "#{n} ha",
-            "square-meter": "#{n} m²",
-            "square-centimeter": "#{n} cm²",
-            "square-mile": "#{n} mi²",
+            "square-kilometer": "#{n} square-kilometer",
```
Comment (ehoogerbeets) [inline js/data/locale/yo/unitfmt.json:11]: The symbol is better than the English identifier in these cases. I would put it back to #{n} km²

File: js/test/date/testdatefmt_yo_BJ.js (hunk 1 above)
Comment (gouniLee) [reply js/test/date/testdatefmt_yo_BJ.js:1327]: No. It's for formatrelative. not duration. so It checks the following keys which CLDR doesn't provide the translation.
```
 "1#1 second ago|#{num} seconds ago": "#-{num} s",
 "1#1 minute ago|#{num} minutes ago": "#-{num} min",
 "1#1 hour ago|#{num} hours ago": "#-{num} h",
 "1#1 day ago|#{num} days ago": "#-{num} d",
 "1#1 week ago|#{num} weeks ago": "#-{num} w",
 "1#1 month ago|#{num} months ago": "#-{num} m",
   ...
```

File: js/test/root/teststrings.js (hunk 4 above)
Comment (gouniLee) [reply js/test/root/teststrings.js]: All value is updated including this test case.

File: js/data/locale/yo/unitfmt.json (hunk 5 above)
Comment (gouniLee) [reply js/data/locale/yo/unitfmt.json]: I updated it.

File: js/test/date/testdatefmt_yo_BJ.js (hunk 1 above)
Comment (ehoogerbeets) [reply js/test/date/testdatefmt_yo_BJ.js:1327]: What mean is that it is odd that we have the translation for the duration formats, but we don't have them for the relative formats. Maybe a special case for this?

Comment (gouniLee) [reply js/test/date/testdatefmt_yo_BJ.js:1327]: That's current CLDR data. Between Duration, and past/fution relative, ilib refers to different clidr file. 
For duration, ilib checks cldr file `cldr-units-full/main/*/units.json`.
But relative, ilib gets data from  cldr file `cldr-dates-full/main/*/dateFields.json`.
For example future year,  cldr-dates-full/main/yo-BJ/dateFiels.json.  it's written below:
```
"relativeTime-type-future": {
     "relativeTimePattern-count-other": "+{0} y"
},

```

**Review by ehoogerbeets — APPROVED** [review]

**Pull request merged**